packages/plugin-node-modules/sources/index.ts

```diff
@@ -6,6 +6,12 @@ import {PnpLooseLinker}                from './PnpLooseLinker';
 
 export const getPnpPath = (project: Project) => ppath.join(project.cwd, `.pnp.js` as Filename);
 
+declare module '@yarnpkg/core' {
+  interface ConfigurationValueMap {
+    nmHoistingLimits: `workspaces` | `dependencies` | `none`;
```

Comment:
```suggestion
    nmHoistingLimits: NodeModulesHoistingLimits;
```

Reply:
Sadly using enums is a no-go. I've added a comment in plugin-essentials as to why, I'll add one here as well

```ts
// Can't use Modifier here because there are actually two instances of this module:
// One in packages/plugin-essentials and one virtual package.
// Defining this property with two different enum instances leads to a compiler error.
```

Reply:
Hmm ... It might still be possible if we use `Record` to provide an indirection (ie instead of typing with the enum, we type with the *values* of the enum). Cf: https://www.typescriptlang.org/play?ts=4.0.2#code/KYOwrgtgBAYg9nAjFA3gKClAglAvFAcgEMCAaDKAIT0ICMy0BfAbjTVElgQCZUKd8xBpmqD65FmwAmwAMYAbIgCdgUCHClh5qggAEAnspAAHANYBzAPSy4Kgn0wBLEABdgSgGZFZqgMJwQD0dzMCUiF0cAgDUieTBgAFkiYwdMKA8EAC4oACU5WykAHngkUigAZxclZ3MAPlZMRiZpOUUVNQ0tHQMjMysbO1TnN09vPwCgkLCI6Nj4pJT0NPSs3PylIpLuMsrqkDqGqCamoA

Reply:
Having an object as type does clash with the rest though. I'm not sure how to tackle this without complicating the `ConfigurationValueMap[K]` syntax